diff --git a/mrbgems/picoruby-ble/ports/esp32/nimble_owner.c b/mrbgems/picoruby-ble/ports/esp32/nimble_owner.c
index cbded16b..779072b7 100644
--- a/mrbgems/picoruby-ble/ports/esp32/nimble_owner.c
+++ b/mrbgems/picoruby-ble/ports/esp32/nimble_owner.c
@@ -228,7 +228,7 @@ pump_service(mrb_state *mrb, void *ud)
   (void)ud;
   mrb_bool error = FALSE;
   int ai = mrb_gc_arena_save(mrb);
-  mrb_protect_error(mrb, pump_body, NULL, &error);
+  pump_body(mrb, NULL);
   mrb_gc_arena_restore(mrb, ai);
 }
 
